portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -43,15 +46,30 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     PreparedQuery results = datastore.prepare(query);
 
     ArrayList<String> comments = new ArrayList<String>();
+    ArrayList<Translation> translatedComments = new ArrayList<Translation>();
+    ArrayList<String> stringOfTranslatedComments = new ArrayList<String>();
```

Comment:
You can write the same code with only one loop and one ArrayList. This saves memory space and makes the code efficient. :)

```
for (Entity entity : results.asIterable()) {
      String text = (String) entity.getProperty("text");

      comments.add(text);
    }
```